v0/destinations/woopra/transform.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-param-reassign */
+const { set } = require("lodash");
+const {
+  defaultRequestConfig,
+  CustomError,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  getErrorRespEvents,
+  getIntegrationsObj,
+  getSuccessRespEvents,
+  defaultGetRequestConfig
+} = require("../../util");
+
+const { EventType } = require("../../../constants");
+const { BASE_URL, mappingConfig, ConfigCategories } = require("./config");
+const { refinePayload } = require("./utils");
+
+const responseBuilder = async (payload, endpoint, method, projectName) => {
+  if (!payload) {
+    throw new CustomError("[ WOOPRA ]:: Parameters could not be found", 400);
+  }
+  set(payload, "Project", projectName);
+  const response = defaultRequestConfig();
+  response.params = removeUndefinedAndNullValues(payload);
+  response.endpoint = endpoint;
+  response.method = method;
+  return response;
+};
+
+const identifyResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/identify`;
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.IDENTIFY.name]
+  );
+  const prefix = ["cv"];
+  const method = defaultGetRequestConfig.requestMethod;
+  refinePayload(message, payload, prefix);
+  const extractedProjectName =
+    getIntegrationsObj(message, "woopra")?.projectName || projectName;
+
+  return responseBuilder(payload, endpoint, method, extractedProjectName);
+};
+const trackResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/ce`;
+  if (!message.event) {
+    throw new CustomError("[ WOOPRA ]:: Event Name can not be empty.", 400);
+  }
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.TRACK.name]
+  );
+  const prefix = ["ce", "cv"];
+  const method = defaultGetRequestConfig.requestMethod;
+  refinePayload(message, payload, prefix);
+  const extractedProjectName =
+    getIntegrationsObj(message, "woopra")?.projectName || projectName;
+  return responseBuilder(payload, endpoint, method, extractedProjectName);
+};
+const pageResponseBuilder = async (message, projectName) => {
+  const endpoint = `${BASE_URL}/ce`;
+  if (!message.name) {
+    throw new CustomError("[ WOOPRA ]:: Page Name can not be empty.", 400);
+  }
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.PAGE.name]
+  );
+  const prefix = ["ce", "cv"];
```

Comment:
same thing here as well. Call the identifyResponseBuilder and get the visitor properties back.

Reply:
Building a common builder as identify builder name will raise confusions and calling that from all the three call functions.